File: 3.py
# https://projecteuler.net/thread=3&page=7#321513
# User: borozcod

num = 600851475143
i = 2
while True:
    if(num % i == 0):
        print(i)
        num = num / i
        print("Number", num)
        # If the resut of the num dived by i is 1 then i must be num, therefore num must be i and largest prime number
        if(num == 1):
            #largest prime factor :)
            print(i)
            break
    else:
        i+=1

File: 3.js
// The prime factors of 13195 are 5, 7, 13 and 29.

// What is the largest prime factor of the number 600851475143 ?

const isPrime = n => {
  for (i = 2; i < n; i++) {
    if (n % i === 0) return;
  }
  return true;
};

// const findLargestPrimeFactor = n => {
//   for (let i = 2; i < n; i++) {
//     if (n % i === 0 && isPrime(n / i)) {
//       return n / i;
//     }
//   }
// };

const findLargestPrimeFactor = n => {
  for (let i = 2; i <= n; i++) {
    if (n % i === 0) {
      n = n / i;
      console.log(i, n);
      if (n === 1) {
        return i;
      }
    }
  }
};

console.log(findLargestPrimeFactor(600851475143));
